### PR TITLE
Update App Javadoc

### PR DESCRIPTION
**Pull request by rorbech**

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 1]
```diff
@@ -51,8 +51,93 @@
 import io.realm.mongodb.functions.Functions;
 
 /**
- * FIXME
+ * An <i>App</i> is the main client side entry point for interacting with a <i>MongoDB Realm App</i>.
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:54]:
```suggestion
 * An <i>App</i> is the main client-side entry point for interacting with a <i>MongoDB Realm App</i>.
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 2]
```diff
@@ -51,8 +51,93 @@
...
+ *
+ * The <i>App</i> can be used to:
+ * <ul>
+ *   <li>Register uses and perform various user related operations through authentication providers
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:58]:
```suggestion
 *   <li>Register uses and perform various user-related operations through authentication providers
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 3]
```diff
@@ -51,8 +51,93 @@
...
+ *   ({@link io.realm.mongodb.auth.ApiKeyAuth}, {@link EmailPasswordAuthImpl})</li>
+ *   <li>Synchronize data between the local device and a remote Realm App with Synchronized Realms</li>
+ *   <li>Invoke Realm App functions with {@link Functions}</li>
+ *   <li>Access remote data from MongoDB databases with a {@link io.realm.mongodb.mongo.MongoClient}</li>
+ * </ul>
+ * <p>
+ * To create an app that is linked with a remote <i>Realm App</i> initialize Realm and configure the
+ * <i>App</i> as below:
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:66]:
```suggestion
 * <i>App</i> as shown below:
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 4]
```diff
@@ -51,8 +51,93 @@
...
+ *             App APP = new App(appConfiguration);
+ *         }
+ *
+ *     }
+ * </pre>
+ * <p>
+ * After configuring the <i>App</i> you can start managing users, configure Synchronized Realms,
+ * call remote Realm Functions and access remote data through Mongo Collections. The below examples
+ * shows the synchronized APIs which cannot be used from the main thread. For the equivalent
+ * asynchronous counter parts. The example project in please see
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:91]: A couple of grammar mistakes.
```suggestion
 * call remote Realm Functions and access remote data through Mongo Collections. The examples below
 * show the synchronized APIs which cannot be used from the main thread. For the equivalent
 * asynchronous counterparts. The example project in please see
```

Also, these two sentences do not make grammatical sense:
```
For the equivalent asynchronous counterparts. The example project in please see
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 5]
```diff
@@ -51,8 +51,93 @@
...
+ * https://github.com/realm/realm-java/tree/v10/examples/mongoDbRealmExample.
+ *
+ * To register a new user and/or login with an existing user as below:
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:94]:
```suggestion
 * To register a new user and/or login with an existing user do as shown below:
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 6]
```diff
@@ -51,8 +51,93 @@
...
+ * <pre>
+ *     // Register new user
+ *     User user = APP.getEmailPasswordAuth().registerUser(username, password);
+ *
+ *     // Login with existing
+ *     APP.login(Credentials.emailPassword(username, password))
+ * </pre>
+ * <p>
+ * With an authorized user you can synchronize data between the local device and the remote Realm
+ * App by opening a Realm with a {@link io.realm.mongodb.sync.SyncConfiguration} as below:
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:104]:
```suggestion
 * App by opening a Realm with a {@link io.realm.mongodb.sync.SyncConfiguration} as indicated below:
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 7]
```diff
@@ -51,8 +51,93 @@
...
+ *     SyncSession session = APP.getSync().getSession(syncConfiguration);
+ *
+ *     instance.executeTransaction(realm -> {
+ *         realm.insert(...);
+ *     });
+ *     session.uploadAllLocalChanges();
+ *     instance.close();
+ * </pre>
+ * <p>
+ * You can call remove Realm functions as below:
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:119]:
```suggestion
 * You can call remove Realm functions as shown below:
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 8]
```diff
@@ -51,8 +51,93 @@
...
+ * <pre>
+ *     Functions functions = user.getFunctions();
+ *     Integer sum = functions.callFunction("sum", Arrays.asList(1, 2, 3, 4), Integer.class);
+ * </pre>
+ * <p>
+ * And access collections from the remote Realm App as below:
```
Comment (edualonso) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:125]: Trying not to abuse the `shown/indicate below`:
```suggestion
 * And access collections from the remote Realm App as shown here:
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 9]
```diff
@@ -51,8 +51,93 @@
...
+ * shows the synchronized APIs which cannot be used from the main thread. For the equivalent
+ * asynchronous counter parts. The example project in please see
+ * https://github.com/realm/realm-java/tree/v10/examples/mongoDbRealmExample.
+ *
+ * To register a new user and/or login with an existing user as below:
+ * <pre>
+ *     // Register new user
+ *     User user = APP.getEmailPasswordAuth().registerUser(username, password);
+ *
+ *     // Login with existing
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:99]:
```suggestion
 *     // Login with existing user
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 10]
```diff
@@ -242,7 +331,10 @@ public User currentUser() {
     }
 
     /**
-     * Switch current user. The current user is the user returned by {@link #currentUser()}.
+     * Switch current user.
+     * <p>
+     *
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:336]:
```suggestion
```

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 11]
```diff
@@ -51,8 +51,93 @@
...
+ *             super.onCreate();
+ *
+ *             Realm.init(this);
+ *
+ *             AppConfiguration appConfiguration = new AppConfiguration.Builder(BuildConfig.MONGODB_REALM_APP_ID)
+ *                     .appName(BuildConfig.VERSION_NAME)
+ *                     .appVersion(Integer.toString(BuildConfig.VERSION_CODE))
+ *                     .build();
+ *
+ *             App APP = new App(appConfiguration);
```
[the remark this answers is not in the document]
Comment (rorbech) [reply realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java]: 👍 Moved

**Pull request merged**